tests/rptest/tests/topic_recovery_test.py

```diff
@@ -942,6 +944,53 @@ def after_restart_validation(self):
             self._produce_and_verify(topic)
 
 
+class AdminApiBasedRestore(FastCheck):
+    def __init__(self, s3_client, kafka_tools, rpk_client: RpkTool, s3_bucket,
+                 logger, rpk_producer_maker, topics, admin: Admin):
+        super().__init__(s3_client, kafka_tools, rpk_client, s3_bucket, logger,
+                         rpk_producer_maker, topics)
+        self.admin = admin
+
+    def _assert_duplicate_request_is_rejected(self):
+        try:
+            # A duplicate request should be rejected as a recovery is already running.
+            response = self.admin.initiate_topic_scan_and_recovery()
+        except requests.exceptions.HTTPError as e:
+            assert e.response.status_code == requests.status_codes.codes[
+                'conflict'], f'request status code: {response.status_code}'
+
+    def _assert_retention(self, value):
+        def wait_for_topic():
+            try:
+                return self._kafka_tools.describe_topic_config(
+                    self.topics[0].name)
+            except:
+                return None
+
+        topic_config = wait_until_result(wait_for_topic, timeout_sec=60)
+        assert topic_config[
+            'retention.local.target.bytes'] == value, f'failed: {topic_config}'
+
+    def _assert_temporary_retention_is_reverted(self):
+        self._assert_retention('-1')
+
+    def restore_redpanda(self, *_):
+        payload = {'retention_ms': 500000}
+        response = self.admin.initiate_topic_scan_and_recovery(payload=payload)
+        assert response.status_code == requests.status_codes.codes[
+            'ok'], f'request status code: {response.status_code}'
+        self._assert_duplicate_request_is_rejected()
+
+    def after_restart_validation(self):
+        super().after_restart_validation()
+        self._assert_temporary_retention_is_reverted()
```

Comment:
I see here that we produce to the topics and make sure that the hwm increases from that produce. I haven't looked around too much but do we also do anything to verify that we can scan from the recreated topic and ensure that we get reasonable data?

Reply:
sorry missed this earlier. We do not have this assertion right now, I will look into adding this in the test soon.